src/inspector/target_agent.cc

```diff
@@ -0,0 +1,136 @@
+#include "target_agent.h"
+#include <string_view>
+#include "crdtp/dispatch.h"
+#include "inspector/worker_inspector.h"
+#include "main_thread_interface.h"
+
+namespace node {
+namespace inspector {
+namespace protocol {
+
+std::unordered_map<int, std::shared_ptr<MainThreadHandle>>
+    TargetAgent::target_session_id_worker_map_ =
+        std::unordered_map<int, std::shared_ptr<MainThreadHandle>>();
+int TargetAgent::next_session_id_ = 1;
+class WorkerTargetDelegate : public WorkerDelegate {
+ public:
+  explicit WorkerTargetDelegate(std::shared_ptr<TargetAgent> target_agent)
+      : target_agent_(target_agent) {}
+
+  void WorkerCreated(const std::string& title,
+                     const std::string& url,
+                     bool waiting,
+                     std::shared_ptr<MainThreadHandle> worker) override {
+    target_agent_->createAndAttachIfNecessary(worker, title, url);
+  }
+
+ private:
+  const std::shared_ptr<TargetAgent> target_agent_;
+};
+
+std::unique_ptr<Target::TargetInfo> createTargetInfo(
+    const std::string_view target_id,
+    const std::string_view type,
+    const std::string_view title,
+    const std::string_view url) {
+  return Target::TargetInfo::create()
+      .setTargetId(std::string(target_id))
+      .setType(std::string(type))
+      .setTitle(std::string(title))
+      .setUrl(std::string(url))
+      .setAttached(false)
+      .setCanAccessOpener(true)
+      .build();
+}
+
+void TargetAgent::Wire(UberDispatcher* dispatcher) {
+  frontend_ = std::make_unique<Target::Frontend>(dispatcher->channel());
+  Target::Dispatcher::wire(dispatcher, this);
+}
+
+void TargetAgent::createAndAttachIfNecessary(
+    std::shared_ptr<MainThreadHandle> worker,
+    const std::string& title,
+    const std::string& url) {
+  std::string target_id = std::to_string(getNextTargetId());
+  std::string type = "worker";
```

Comment:
This type conflicts with the web worker type in https://source.chromium.org/chromium/chromium/src/+/main:third_party/devtools-frontend/src/front_end/core/sdk/ChildTargetManager.ts;l=176?q=ChildTargetManager.ts

If devtools-frontend recognize Node.js worker as a web worker, it will fail to attach to this target as Node.js does not support all domains supported by a web worker.

Reply:
> Would you like to draft the change as well? Otherwise I can help with the work to land the change in devtools-frontend.

Thx, I have prepared a draft on my end.
https://chromium-review.googlesource.com/c/devtools/devtools-frontend/+/6218804

For now, I named the new type node_worker.